From 8eb667782468f42b15b6b35975a26b03042599c1 Mon Sep 17 00:00:00 2001
From: dianlujitao <dianlujitao@lineageos.org>
Date: Sat, 2 Feb 2019 21:08:03 +0800
Subject: [PATCH 25/36] Keyguard: Avoid starting FP authentication right after
 cancelling

 * SystemUI would start FP authentication right after unlock in non-FP
   ways (e.g., pattern, PIN) after e22f1347e8da99676f68864927acbd148dc430a5,
   obviously this is redundant, and can cause unexpected behavior if the
   FP sensor has relatively high latency.

Change-Id: Iaa19c3113408d3119b4aebb2b08e9922739267ed
---
 .../SystemUI/src/com/android/keyguard/KeyguardUpdateMonitor.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/packages/SystemUI/src/com/android/keyguard/KeyguardUpdateMonitor.java b/packages/SystemUI/src/com/android/keyguard/KeyguardUpdateMonitor.java
index 04206b9d4a0..71f3ec62295 100644
--- a/packages/SystemUI/src/com/android/keyguard/KeyguardUpdateMonitor.java
+++ b/packages/SystemUI/src/com/android/keyguard/KeyguardUpdateMonitor.java
@@ -1291,6 +1291,7 @@ public class KeyguardUpdateMonitor implements TrustManager.TrustListener,
         if (mFingerprintRunningState == FINGERPRINT_STATE_RUNNING && !shouldListenForFingerprint) {
             stopListeningForFingerprint();
         } else if (mFingerprintRunningState != FINGERPRINT_STATE_RUNNING
+                && mFingerprintRunningState != FINGERPRINT_STATE_CANCELLING
                 && shouldListenForFingerprint) {
             startListeningForFingerprint();
         }
-- 
2.17.1

